plumbing/reference.go

```diff
@@ -222,3 +222,11 @@ func (r *Reference) String() string {
 	v.WriteString(name)
 	return v.String()
 }
+
+// ReferenceSlice attaches the methods of sort.Interface to []*Reference, sorting in
+// increasing order.
+type ReferenceSlice []*Reference
+
+func (p ReferenceSlice) Len() int           { return len(p) }
+func (p ReferenceSlice) Less(i, j int) bool { return p[i].Name() < p[j].Name() }
+func (p ReferenceSlice) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
```

Comment:
This seems to be orthogonal to the PR's objective. Do you mind splitting this into its own PR - with some tests.

```suggestion
```